tools/sched_ext/scx_serialise.bpf.c

```diff
@@ -375,57 +360,114 @@ static u32 get_executor_id(const struct task_struct *p) {
 	return eid;
 }
 
+static int update_priority(pid_t pid, struct task_ctx* tctx) {
+	bpf_spin_lock(&tctx->lock);
+
+	s32 priority = assign_priority(pid);
+	if (priority < 0) {
+		bpf_spin_unlock(&tctx->lock);
+		return -1;
+	}
+
+	tctx->priority = priority;
+	tctx->enqueued = true;
+	bpf_spin_unlock(&tctx->lock);
+	return 0;
+}
+
+static long update_priority_callback(struct bpf_map *map, const pid_t *pid, struct task_ctx *tctx, const u32 *eid) {
```

Comment:
This is not implemented correctly. 

The "assign priority" function takes in the current task id, but must consider *all* priorities at once. This is why it was called "update all priorities" in the previous version of the code.

The reason for this is because different algorithms may update only some subset of priorities relevant to the current task. E.g. RP only should update the current task's priority, but the way this code is written, it will update all tasks.